codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectAggregator.java

```diff
@@ -151,7 +151,11 @@ protected boolean isAggregated(HttpObject msg) throws Exception {
 
     @Override
     protected boolean isContentLengthInvalid(HttpMessage start, int maxContentLength) {
-        return getContentLength(start, -1L) > maxContentLength;
+        try {
+            return getContentLength(start, -1L) > maxContentLength;
+        } catch (final NumberFormatException e) {
+            return false;
```

Comment:
`invalid` can have multiple meanings in this context ... check out the javadoc for more clarity.